### DISCUSSION TITLE
Using private channel whithout '$request->user()' but with JWT

### DISCUSSION THREAD

**Discussion by fabioselau077** [laravel/framework#49128] [Q&A]
### Laravel Version

10.10

### PHP Version

8.3

### Description

Hello, I'm configuring Laravel with self hosted pusher (soketi), everything works for public channels. However, when it is a private channel (private-), the call is made to site.com/api/broadcasting/auth, but 403 is always returned.

Analyzing the Laravel source code and debugging ( https://prnt.sc/JlUUq0DvyRxk), I verified a check in the PusherBroadcaster auth function that tries to retrieve the user() within retrieveUser(), but as I use JWT I don't have "$request->user()", after passing through the middleware 'verify.jwt' and I verify the JWT, I decrypt it in Bearer and I add it to $request->credentials (https://prnt.sc/lOL0gFz3Lwrf).

I believe this is where the problem lies, even though we don't say to use any middleware or anything like that, it always waits for $request->user() because it has 'private-' in the channel name.

The main question is: Is there a way to avoid falling into these validations? Because if the $request falls into the routes/channels, I can validate it through the JWT and define whether or not it has access to that channel, however it gives 403 before arriving, because the $request->user() is analyzed, which In my case, I don't use Passport or another Laravel authentication method because I use it as a 100% API.

Any solution?

### Steps To Reproduce

1. Create new project
2. Configure one middleware JWT
3. Configure broadcasters for Pusher
4. Call site.com/broadcasting/auth and receveid 403

**Comment by ghanu119** [comment]
@fabioselau077 
Have you checked out this configuration?
Is it not helpful?

[[Customizing The Authorization Endpoint](https://laravel.com/docs/10.x/broadcasting#customizing-the-authorization-endpoint)](https://laravel.com/docs/10.x/broadcasting#customizing-the-authorization-endpoint)
